tests/test_default_executor.py

```diff
@@ -0,0 +1,238 @@
+from typing import Any
+
+from _pytest.capture import CaptureResult
+
+from kraken.core.system.executor.default import (
+    DefaultGraphExecutor,
+    DefaultPrintingExecutorObserver,
+    DefaultTaskExecutor,
+)
+from kraken.core.system.graph import TaskGraph
+from kraken.core.system.project import Project
+from kraken.core.system.task import Task
+
+
+class MyTask(Task):
+    """
+    Fake task
+    """
+
+    def execute(self) -> None:
+        print("Hello")
+
+
+class MyFailingTask(Task):
+    """
+    Fake failing task
+    """
+
+    def execute(self) -> None:
+        raise RuntimeError("Wow this is failing")
+
+
+skipped_tests_title = "Tasks that were not executed due to failing dependencies"
+
+
+def execute_print_test(graph: TaskGraph) -> None:
+    """
+    Basic common execution to get the final printed result
+    """
+    default_task_executor = DefaultTaskExecutor()
+    default_printing_executor_observer = DefaultPrintingExecutorObserver()
+    default_executor = DefaultGraphExecutor(default_task_executor)
+    default_executor.execute_graph(graph, default_printing_executor_observer)
+
+
+def trim_printed_result(captured: CaptureResult[str]) -> str:
+    """
+    Retrieving the skipped test printed part
+    """
+    print_output = str(captured.out)
+    title_index_end = print_output.find(skipped_tests_title) + len(skipped_tests_title)
+    result = print_output[(title_index_end + 1) :]
+    return result
+
+
+def test__DefaultExecutor__print_correct_failures_with_dependencies(kraken_project: Project, capsys: Any) -> None:
+    """This test tests if when a task failed, successor tasks with depedencies will be printed as failed.
+
+    ```
+    A
+    |
+    v
+    B
+    |
+    v
+    C
+    |
+    v
+    D
+    ```
+    If B fails, C, D should be printed.
+    """
+    task_a = kraken_project.do("a", MyTask)
+    task_b = kraken_project.do("b", MyFailingTask)
+    task_c = kraken_project.do("c", MyTask)
+    task_d = kraken_project.do("d", MyTask)
+
+    task_b.add_relationship(task_a)
+    task_c.add_relationship(task_b)
+    task_d.add_relationship(task_c)
+
+    graph = TaskGraph(kraken_project.context).trim([task_d])
+    assert set(graph.tasks()) == {task_b, task_c, task_a, task_d}
```

Comment:
not a big deal, but we might as well sort them alphabetically

```suggestion
    assert set(graph.tasks()) == {task_a, task_b, task_c, task_d}
```

Reply:
Fixed